src/components/fields/hidden-field/types.ts

```diff
@@ -1,6 +1,38 @@
+import { IYupValidationRule } from "../../types";
 import { IBaseFieldSchema } from "../types";
 
-export interface IHiddenFieldSchema<V = undefined>
-	extends Pick<IBaseFieldSchema<"hidden-field", V>, "showIf" | "validation" | "uiType"> {
-	valueType?: "string" | "number" | "boolean" | undefined;
+export interface IHiddenFieldValidationRule extends IYupValidationRule {
+	/** for customising error message when submitted value does not match schema value */
+	matchesSchema?: boolean | undefined;
```

Comment:
```suggestion
	equalsSchemaValue?: boolean | undefined;
```
I think we can keep to the `equals` prefix similar to `equals` and `equalsField` rules and be more specific because I was under the impression it is meant to match a schema structure.

other than that, why are we applying the validation in the validation-schema-generator only? the rule becomes useless if the validation-schema-generator is not used

Reply:
the hidden field always overrides the form value to match the schema value, hence I didn't expect any validation required on the frontend

Reply:
actually, under what circumstances will the submitted value differ from the schema value? shouldnt they always match? i.e. this rule is not necessary and should be inbuilt rather than declared by the consumer?

Reply:
this is a safeguard for backend submission, in case the request body gets overridden

by inbuilt do you mean a generic error message? this rule is for anyone who may need to customise the error message on the backend

Reply:
I mean if `value` is declared in the schema, it should be implied that the submitted value will be equal to that value, hence any validation should be applied automatically (in validation-schema-generator). so the `equalsSchemaValue` rule is not necessary after all.

Reply:
oh right, I forgot that this is merely for customising the error message